Modules/Filtering/ImageFilterBase/include/itkCastImageFilter.h

```diff
@@ -119,22 +123,33 @@ class ITK_TEMPLATE_EXPORT CastImageFilter:
 
 #ifdef ITK_USE_CONCEPT_CHECKING
   // Begin concept checking
-  itkConceptMacro( InputConvertibleToOutputCheck,
-                   ( Concept::Convertible< typename TInputImage::PixelType,
-                                           typename TOutputImage::PixelType > ) );
+  //static_assert(std::is_convertible<InputPixelType, OutputPixelType>::value );
```

Comment:
Did you intend for this to be commented out?

Reply:
This patch gets RGB->Vector pixel types to be "casted" or converted by the filter. Those types are not convertible. But `InputPixelType::ValueType`->`OutputPixelType::ValueType` is... but for scalar don't have the `ValueType` typedef. 

@N-Dekker 
I guess what's needed is a meta-programming "trait" that defaults to fail when SFINAE. for 

```
std::is_convertible<typename InputPixelType::ValueType, typename OutputPixelType::ValueType>::value 
```

Then we can just or the two trait's values.

Reply:
Then perhaps remove this commented code?